scipy/stats/_continuous_distns.py

```diff
@@ -6898,6 +6910,40 @@ def _ppf(self, q, c, d):
                       1 - np.sqrt((1 - q) * (d - c + 1) * (1 - d))]
         return np.select(condlist, choicelist)
 
+    def _munp(self, n, c, d):
+        # Using the parameterization from Kracker, et al., with
+        # a=bottom left, c=top left, d=top right, b=bottom right, then
+        #     E[X^n] = h/(n+1)/(n+2) [(b^{n+2}-d^{n+2})/(b-d)
+        #                             - ((c^{n+2} - a^{n+2})/(c-a)]
+        # with h = 2/((b-a) - (d-c)). The corresponding parameterization
+        # in scipy, has a'=loc, c'=loc+c*scale, d'=loc+d*scale, b'=loc+scale,
+        # which for standard form reduces to a'=0, b'=1, c'=c, d'=d.
+        # Substituting into E[X^n] gives the bd' term as (1 - d^{n+2})/(1 - d)
+        # and the ac' term as c^{n-1} for the standard form. The bd' term has
+        # numerical difficulties near d=1, so replace (1 - d^{n+2})/(1-d)
+        # with expm1((n+2)*log(d))/(d-1).
```

Comment:
I didn't mean to ask about `expm1` specifically; I meant was there a reference for the trick to replace a simple fraction with an exponential of a log. I see that they are equivalent mathematically, but it is not immediately obvious why the latter is more stable.

Reply:
The replacement is only in the numerator, where `d^{n+2} - 1` goes to `expm1((n+2)*log(d))` to avoid the cancellation error when `d ≈ 1`.  The denominator `(d-1)` also has cancellation, but the power in the numerator dominates.  I checked `d = 1-eps` for various `n` from 0 to 20 to confirm this.

Reply:
OK, thanks.